File: csharp/剑指21调整数组顺序使奇数位于偶数前面.cs
public partial class Solution
{
    public int[] Exchange(int[] nums)
    {
        int i, j;
        (i, j) = (0, nums.Length-1);

        while (i < j)
        {
            for (; i < j && nums[i] % 2 == 1; i++) ;
            for (; i < j && nums[j] % 2 == 0; j--) ;

            if(i<j)
                (nums[i], nums[j]) = (nums[j], nums[i]);
            
            
        }

        return nums;
    }
}

File: python/main/剑指21调整数组顺序使奇数位于偶数前面.py

def exchange(nums: list) -> list:
    i,j=0,len(nums)-1

    while(i<j):

        while(True):
            if(i<j and nums[i]%2==1):
                i+=1
            else:
                break
        
        while(True):
            if(i<j and nums[j]%2==0):
                j-=1
            else:
                break
        
        if(i<j):
            nums[i],nums[j]=nums[j],nums[i]
            i+=1
            j-=1
    
    return nums

print(exchange([1,2,1,1]))